doc/rvv-intrinsic-spec.adoc

```diff
@@ -0,0 +1,523 @@
+| uint32_t  | N/A          | N/A          | *vuint32mf2_t* | vuint32m1_t   | vuint32m2_t   | vuint32m4_t | vuint32m32_t
+| uint64_t  | N/A          | N/A          | N/A           | *vuint64m1_t* | *vuint64m2_t* | *vuint64m4_t* | *vuint64m8_t*
+|===
+
+[[floating-point-type]]
+=== Floating-point types
+
+The floating-point types have SEW and LMUL encoded into the type. The first row describes the LMUL and the first column describes the data type and element width of the scalar type.
+
+Floating-point types with element widths of 16 requires the `zvfh` extension to be specified in the architecture.
+
+Floating-point types with element widths of 32 requires the `zve32f` extension to be specified in the architecture.
+
+Floating-point types with element widths of 64 requires the `zve64d` extension to be specified in the architecture.
+
+.Floating-point types
+[cols="2,2,2,2,2,2,2,2"]
+[%autowidth]
+|===
+| Types      | LMUL=1/8 | LMUL=1/4   | LMUL=1/ 2   | LMUL=1     | LMUL=2     | LMUL=4   | LMUL=8
+| float16_t  | N/A        | vfloat16m4_t | vfloat16mf2_t | vfloat16m1_t | vfloat16m2_t | vfloat16m4_t | vfloat16m16_t
+| float32_t  | N/A        | N/A          | vfloat32mf2_t | vfloat32m1_t | vfloat32m2_t | vfloat32m4_t | vfloat32m32_t
+| float64_t  | N/A        | N/A          | N/A           | vfloat64m1_t | vfloat64m2_t | vfloat64m4_t | vfloat64m8_t
+|===
+
+=== Mask types
+
+The mask types encode the ratio that is derived from `SEW` / `LMUL`. The mask types represent mask register values that follows the Mask Register Layou ^15^t.
+
+.Mask types
+[cols="2,2,2,2,2,2,2,2"]
+[%autowidth]
+|===
+| Types | n = 1    | n = 2    | n = 4    | n = 8    | n = 16    | n = 32    | n = 64
+| bool  | vbool1_t | vbool2_t | vbool4_t | vbool8_t | vbool16_t | vbool32_t | vbool64_t
+|===
+
+=== Tuple type
+
+The tuple types encodes `SEW`, `LMUL`, and `NFIELD`^17^ into the data type.
+
+These types are utilized for the segment load/store instruction intrinsics, the types listed in <<integer-type>> and <<floating-point-type>> all have tuple types. Types under the combination of `LMUL`, `NFIELD` follows the restiction by v-spec - EMUL * NFIELDS ≤ 8.
+
+The full list of types is attached in the Appendix.
+
+[[chapter-pseudo-instruction]]
+== Pseudo instructions
+
+The intrinsics provide extra utility functions to help users manipulate across the RVV intrinsic types. These functions are called "pseudo intrinsics". These pseudo intrinsics do not represent any real instructions.
+
+[[pseudo-vsetvl-vsetvlmax]]
+=== `vsetvl` / `vsetvlmax`
+
+[[pseudo-vreinterpret]]
+=== `vreinterpret`
+
+[[pseudo-vundefined]]
+=== `vundefined`
+
+[[pseudo-vget-vset]]
+=== `vget` / `vset`
+
+[[pseudo-vlmul_trunc-vlmul_ext]]
+=== `vlmul_trunc` / `vlmul_ext`
+
+== Programming Notes
+
+=== Builtin mapping
+
+Builtin mapping of the intrinsics is a compiler and system specific issue.
```

Comment:
Not sure this is worth mentioning

Reply:
My thought that is that some users may expect the intrinsics to directly translate and appear in the assembly, but the compiler may optimize them out if there is an opportunity. That is why I added this side note.

Reply:
Maybe you should say that instead of mentioning "builtin"? I read this as referring to the naming of the C intrinsics vs the __builtin names that they alias.

Reply:
Yes I agree. Just rephrased them.